src/pkg/sources/remote.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	zarfTypes "github.com/defenseunicorns/zarf/src/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content/file"
+
+	"github.com/defenseunicorns/uds-cli/src/config"
+)
+
+// RemoteBundle is a package source for remote bundles that implements Zarf's packager.PackageSource
+type RemoteBundle struct {
+	PkgName        string
+	PkgOpts        *zarfTypes.ZarfPackageOptions
+	PkgManifestSHA string
+	TmpDir         string
+	Remote         *oci.OrasRemote
+}
+
+// LoadPackage loads a Zarf package from a remote bundle
+func (r *RemoteBundle) LoadPackage(dst *layout.PackagePaths, unarchiveAll bool) error {
+	packageSpinner := message.NewProgressSpinner("Loading bundled Zarf package: %s", r.PkgName)
+	defer packageSpinner.Stop()
+	err := r.downloadPkgFromRemoteBundle()
+	if err != nil {
+		return err
+	}
+
+	var files []string
+	var pkg zarfTypes.ZarfPackage
+	err = filepath.Walk(dst.Base, func(path string, fileInfo fs.FileInfo, err error) error {
+		if err != nil {
+			return nil
+		}
+		if fileInfo.IsDir() {
+			return nil
+		}
+		newString := strings.ReplaceAll(path, dst.Base+"/", "")
+		files = append(files, newString)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	if err = utils.ReadYaml(dst.ZarfYAML, &pkg); err != nil {
+		return err
+	}
+
+	dst.SetFromPaths(files)
+
+	if unarchiveAll {
+		for _, component := range pkg.Components {
+			if err := dst.Components.Unarchive(component); err != nil {
+				if layout.IsNotLoaded(err) {
+					_, err := dst.Components.Create(component)
+					if err != nil {
+						return err
+					}
+				} else {
+					return err
+				}
+			}
+		}
+
+		if dst.SBOMs.Path != "" {
+			if err := dst.SBOMs.Unarchive(); err != nil {
+				return err
+			}
+		}
+	}
+	packageSpinner.Successf("Loaded bundled Zarf package: %s", r.PkgName)
+	return nil
+}
+
+// LoadPackageMetadata loads a Zarf package's metadata from a remote bundle
+func (r *RemoteBundle) LoadPackageMetadata(dst *layout.PackagePaths, _ bool, _ bool) (err error) {
+	root, err := r.Remote.FetchRoot()
+	if err != nil {
+		return err
+	}
+	var zarfManifest *oci.ZarfOCIManifest
+	for _, layer := range root.Layers {
+		if layer.Digest.Encoded() == r.PkgManifestSHA {
+			zarfManifest, err = r.Remote.FetchManifest(layer)
+			if err != nil {
+				return err
+			}
+			break
+		}
+	}
+
+	if zarfManifest == nil {
+		return fmt.Errorf(fmt.Sprintf("Zarf image manifest with sha %s not found", r.PkgManifestSHA))
+	}
+
+	for _, layer := range zarfManifest.Layers {
+		if layer.Annotations[ocispec.AnnotationTitle] == config.ZarfYAML {
+			zarfYamlBytes, err := r.Remote.FetchLayer(layer)
+			if err != nil {
+				return err
+			}
+			if err := os.WriteFile(filepath.Join(dst.Base, config.ZarfYAML), zarfYamlBytes, 0644); err != nil {
+				return err
+			}
+			break
+		}
+	}
+
+	return nil
+}
```

Comment:
This can be simplified to something like:

```
root, err := r.Remote.FetchRoot()
...
manifestDesc := root.Locate(r.PkgManifestSha)
...
if oci.IsEmptyDescriptor(manifestDesc) // handle
...
manifest, err := r.remote.FetchManifest(manifestDesc)
...
pkg, err := oci.FetchZarfYaml(manifest)
...
if err := utils.WriteYaml(pkg) // dunno if this exists 
...
```